executor/analyze.go

```diff
@@ -181,32 +181,38 @@ func filterAndCollectTasks(tasks []*analyzeTask, statsHandle *handle.Handle, inf
 		tids                = make([]int64, 0, len(tasks))
 		// taskMap is used to collect the tasks that belong to the same table.
 		// In stats v1, analyze for each index is a single task, and they have the same table id.
-		taskMap = make(map[int64][]*analyzeTask, len(tasks))
+		tidMap = make(map[int64]struct{}, len(tasks))
 	)
 
+	// Check the locked tables in one transaction.
 	for _, task := range tasks {
 		tableID := getTableIDFromTask(task)
 		tids = append(tids, tableID)
-		taskMap[tableID] = append(taskMap[tableID], task)
 	}
-
-	// Check the locked tables in one transaction.
-	lockedStatuses, err := statsHandle.QueryTablesLockedStatuses(tids...)
+	lcokedTables, err := statsHandle.GetLockedTables(tids...)
 	if err != nil {
 		return nil, 0, nil, err
 	}
 
-	for tid, isLocked := range lockedStatuses {
+	for _, task := range tasks {
+		tableID := getTableIDFromTask(task)
+		_, isLocked := lcokedTables[tableID]
 		if !isLocked {
-			filteredTasks = append(filteredTasks, taskMap[tid]...)
-			needAnalyzeTableCnt++
-		} else {
-			tbl, ok := infoSchema.TableByID(tid)
-			if !ok {
-				logutil.BgLogger().Warn("Unknown table ID in analyze task", zap.Int64("tid", tid))
+			filteredTasks = append(filteredTasks, task)
+		}
+		if _, ok := tidMap[tableID]; !ok {
+			if isLocked {
+				tbl, ok := infoSchema.TableByID(tableID)
+				if !ok {
+					// Ignore this table because it may have been dropped.
```

Comment:
Looks like this line is mistakenly added back.

Reply:
Removed.